### ISSUE TITLE
Patch 2.0.1

### ISSUE THREAD

**Issue by tjayharvey** [GregRos/IEMod.pw#38]
Causes no compile errors. Can boot game successfully. Will test mods as I play.

Tested:
- [x] CipherFocus
- [x] PerEncounterSpells
- [x] FastSneak
- [x] ToggleWalk
- [x] Console
- [x] Bonus Spells in Grimoire
- [ ] GameSpeed
- [ ]  Always ShowCircles
- [ ] Targeting
- [ ] OneTooltip
- [x] Bonus Spells per Day
- [ ] Options
- [ ] Camera 
- [ ] Experience
- [x] Backer Names
- [x] Blue Circles (±IE Blue)
- [ ] CombatOnly (spells)
- [ ] CombatLooting
- [ ] Game Object Browser
- [ ] NoEngagement
- [x] UI Customization

**Comment by Bigtuna00** [comment]
Are you using 5.0.2 or did you do a local build? I can't load any savegames.

**Comment by tjayharvey** [comment]
I did a local build after patching.

It's 100% safe to assume that every patch will break the existing binary. The question is whether it breaks anything in the actual mod. If not, it's just "push a button to generate the new binaries".

**Comment by tjayharvey** [comment]
Added some further items I've tested.

Not 100% sure how to test some of the rest.

**Comment by GregRos** [comment]
I'll run a diff with the old code and see if anything meaningful has changed

**Comment by GregRos** [comment]
Okay, I did it (had to change my tactics a bit this time, but I got it to work well in the end). I ruined my managed folder and I can't test it though.

There were several small, niggling changes throughout the code. It makes me think that distributing a patcher that lets anyone patch their new version with an outdated mod is probably a bad idea. I mean, a patcher is great (in fact, I'm working on something like that now, will post some conceptual work soon), but it should also check that the game version is compatible with the mod, and if not, just fail to work, unless you do something technical like supplying a command-line switch (basically a way of asking _are you absolutely sure you want to do this?_)

Otherwise, people will experience random bugs, or fixes to previous bugs would fail to work because they are overwritten by outdated mod code.

Maybe we can figure out a way to transform code in a way that preserves the original (with more power than `DuplicatesBody` allows). That will make updating the mod much easier.
